pkg/registry/file/dynamicpathdetector/analyze_endpoints.go

```diff
@@ -0,0 +1,147 @@
+package dynamicpathdetector
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	types "github.com/kubescape/storage/pkg/apis/softwarecomposition"
+)
+
+func AnalyzeEndpoints(endpoints *[]types.HTTPEndpoint, analyzer *PathAnalyzer) ([]types.HTTPEndpoint, error) {
+	var newEndpoints []types.HTTPEndpoint
+	MergeDuplicateEndpoints(endpoints)
+	for _, endpoint := range *endpoints {
+		AnalyzeURL(endpoint.Endpoint, analyzer)
```

Comment:
I call it here and later because the first call already detect the dynamic segment and the second call is to check if the url is dynamic.
This is to prevent a case when if we're doing one loop we won't detect the first 100.
I can develop a cache mechanism for AnalyzeURL

Reply:
let's leave it this way